### ISSUE TITLE
nextcloud log: cron

### ISSUE THREAD

**Issue by jookk** [nextcloud/server#13688]
### Steps to reproduce
1.
2.
3.

### Expected behaviour
no error in log

### Actual behaviour
TypeError: Return value of OCP\BackgroundJob\Job::getId() must be of the type integer, string returned

    /var/www/nextcloud/cron.php - line 118:

    OCP\BackgroundJob\Job->getId()

### Server configuration

**ubuntu 18.04 lts**:

**nginx**

**mariadb**

**7.2.10**

**15.0.2:** (see Nextcloud admin page)

**Updated from an older Nextcloud/ownCloud or fresh install:**
updated
**Where did you install Nextcloud from:**
webpage nextcloud zip
**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 

No errors have been found. 
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.1.0
  - activity: 2.8.2
  - apporder: 0.6.0
  - audioplayer: 2.5.1
  - bruteforcesettings: 1.3.0
  - calendar: 1.6.4
  - cloud_federation_api: 0.1.0
  - comments: 1.5.0
  - contacts: 3.0.1
  - dav: 1.8.1
  - federatedfilesharing: 1.5.0
  - federation: 1.5.0
  - files: 1.10.0
  - files_downloadactivity: 1.4.0
  - files_pdfviewer: 1.4.0
  - files_rightclick: 0.10.1
  - files_sharing: 1.7.0
  - files_texteditor: 2.7.0
  - files_trashbin: 1.5.0
  - files_versions: 1.8.0
  - files_videoplayer: 1.4.0
  - firstrunwizard: 2.4.0
  - gallery: 18.2.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.3.0
  - news: 13.0.3
  - nextcloud_announcements: 1.4.0
  - notes: 2.5.1
  - notifications: 2.3.0
  - oauth2: 1.3.0
  - password_policy: 1.5.0
  - passwords: 2019.1.0
  - phonetrack: 0.4.1
  - piwik: 0.5.0
  - provisioning_api: 1.5.0
  - ransomware_protection: 1.3.0
  - serverinfo: 1.5.0
  - sharebymail: 1.5.0
  - sharerenamer: 2.3.0
  - spreed: 5.0.0
  - support: 1.0.0
  - theming: 1.6.0
  - twofactor_backupcodes: 1.4.1
  - twofactor_nextcloud_notification: 1.0.2
  - twofactor_totp: 2.1.0
  - updatenotification: 1.5.0
  - workflowengine: 1.5.0
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nxxx.es"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/nxxx.es",
        "dbtype": "mysql",
        "version": "15.0.2.0",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "defaultapp": "apporder",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "maintenance": false,
        "theme": "",
        "log_type": "file",
        "logtimezone": "Europe\/xxx",
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "loglevel": 2,
        "updater.release.channel": "stable",
        "enabledPreviewProviders": [
            "OC\\Preview\\Image",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\Epub",
            "OC\\Preview\\PDF"
        ],
        "allowNightlyUpdates": [],
        "mail_sendmailmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_smtpauth": 1,
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***"
    }
}

</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no
#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**
firefox
**Operating system:**
win 10
#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"oed3K49WMkxrgyd2dFZg","level":2,"time":"2019-01-17T19:12:26+01:00","remoteAddr":"xxx","user":"--","app":"core","method":"POST","url":"\/login?redi$
{"reqId":"52UhoGxSKly8x4xcQTiE","level":2,"time":"2019-01-17T19:12:34+01:00","remoteAddr":"xxx","user":"--","app":"core","method":"POST","url":"\/login?redi$
{"reqId":"6QiVEAr7083g99DnFzRA","level":3,"time":"2019-01-18T04:15:04+01:00","remoteAddr":"","user":"--","app":"cron","method":"","url":"--","message":{"Exception":"Ty$
{"reqId":"k63PU03vV9ZMrhel0qc6","level":3,"time":"2019-01-18T10:12:21+01:00","remoteAddr":"xxx","user":"usernameP","app":"notifications","method":"POST","ur$
{"reqId":"R8ADeaStdIMLtGMYxxr3","level":3,"time":"2019-01-18T16:15:04+01:00","remoteAddr":"","user":"--","app":"cron","method":"","url":"--","message":{"Exception":"Ty$
{"reqId":"TI7JSOOWVsA8rMkszb6U","level":3,"time":"2019-01-18T20:37:30+01:00","remoteAddr":"xxx","user":"user2","app":"notifications","method":"POST","url":"\$
{"reqId":"K1axgggONVtlfDhFNH81","level":3,"time":"2019-01-18T20:37:49+01:00","remoteAddr":"xxx","user":"user2","app":"notifications","method":"POST","url":"\$
{"reqId":"3G1Sqm195emH5vDpIzVh","level":3,"time":"2019-01-18T20:38:18+01:00","remoteAddr":"xxx","user":"user2","app":"notifications","method":"POST","url":"\$
{"reqId":"fE1AKXq54pacCJWMDqZu","level":3,"time":"2019-01-18T20:38:34+01:00","remoteAddr":"xxx","user":"user2","app":"notifications","method":"POST","url":"\$
{"reqId":"PJpAfjYMkmo83Pf3mVs1","level":3,"time":"2019-01-18T20:39:00+01:00","remoteAddr":"xxx","user":"user2","app":"notifications","method":"POST","url":"\$
{"reqId":"m3ypJ7L7UWqvDEQ7SxIG","level":3,"time":"2019-01-18T20:39:19+01:00","remoteAddr":"xxx","user":"user2","app":"notifications","method":"POST","url":"\$
{"reqId":"87pmF5benpMHzyfQYm86","level":3,"time":"2019-01-18T20:44:34+01:00","remoteAddr":"xxx","user":"user1admin","app":"notifications","method":"POST","url":$
{"reqId":"LbEEc0rTFA6oelHeH5NJ","level":3,"time":"2019-01-19T04:15:04+01:00","remoteAddr":"","user":"--","app":"cron","method":"","url":"--","message":{"Exception":"Ty$
{"reqId":"IoHT1Azx67KWT5qVDkP7","level":3,"time":"2019-01-19T10:43:23+01:00","remoteAddr":"xxx","user":"user2","app":"notifications","method":"POST","url":"\$


```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by violoncelloCH** [comment]
cc @nextcloud/server-triage

**Comment by MorrisJobke** [comment]
> {"reqId":"LbEEc0rTFA6oelHeH5NJ","level":3,"time":"2019-01-19T04:15:04+01:00","remoteAddr":"","user":"--","app":"cron","method":"","url":"--","message":{"Exception":"Ty$

Would it be possible to see the full line here?

**Comment by violoncelloCH** [comment]
```
{"reqId":"xOE0ATQhfjxJRX4ly24x","level":3,"time":"2019-01-21T12:00:01+00:00","remoteAddr":"","user":"--","app":"cron","method":"","url":"--","message":{"Exception":"TypeError","Message":"Return value of OCP\\BackgroundJob\\Job::getId() must be of the type integer, string returned","Code":0,"Trace":[{"file":"\/var\/www\/html\/nextcloud\/cron.php","line":118,"function":"getId","class":"OCP\\BackgroundJob\\Job","type":"->","args":[]}],"File":"\/var\/www\/html\/nextcloud\/lib\/public\/BackgroundJob\/Job.php","Line":117,"CustomMessage":"--"},"userAgent":"--","version":"15.0.2.0"}
```
that's what is in my log...

@MorrisJobke sorry, I've overlooked that the originally posted log is truncated...

**Comment by jookk** [comment]
{"reqId":"LbEEc0rTFA6oelHeH5NJ","level":3,"time":"2019-01-19T04:15:04+01:00","remoteAddr":"","user":"--","app":"cron","method":"","url":"--","message":{"Exception":"TypeError","Message":"Return value of OCP\\BackgroundJob\\Job::getId() must be of the type integer, string returned","Code":0,"Trace":[{"file":"\/var\/www\/nextcloud.es\/cron.php","line":118,"function":"getId","class":"OCP\\BackgroundJob\\Job","type":"->","args":[]}],"File":"\/var\/www\/nextcloud.es\/lib\/public\/BackgroundJob\/Job.php","Line":117,"CustomMessage":"--"},"userAgent":"--","version":"15.0.2.0"}

**Comment by MorrisJobke** [comment]
Could you check what in the table `oc_jobs` is stored? Maybe there are more details about this issue.

**Comment by jookk** [comment]
I like do it, but what do you want to see? Can I place screenshot?
Thx for help :)

Dňa 22. januára 2019 16:14:32 používateľ Morris Jobke 
<notifications@github.com> napísal:

**Comment by violoncelloCH** [comment]
@MorrisJobke here, the content of the table from my production instance:
<details>

```
MariaDB [(none)]> SELECT * FROM nextcloud.oc_jobs;
+-------+-----------------------------------------------------------------+----------------------------------------------------------------+------------+--------------+-------------+--------------------+
| id    | class                                                           | argument                                                       | last_run   | last_checked | reserved_at | execution_duration |
+-------+-----------------------------------------------------------------+----------------------------------------------------------------+------------+--------------+-------------+--------------------+
|     1 | OCA\Files_Trashbin\BackgroundJob\ExpireTrash                    | null                                                           | 1548168302 |   1548170102 |           0 |                  0 |
|     2 | OCA\Files\BackgroundJob\ScanFiles                               | null                                                           | 1548170101 |   1548170102 |           0 |                  1 |
|     3 | OCA\Files\BackgroundJob\DeleteOrphanedItems                     | null                                                           | 1548166502 |   1548170102 |           0 |                  0 |
|     4 | OCA\Files\BackgroundJob\CleanupFileLocks                        | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
|     5 | OCA\DAV\CardDAV\Sync\SyncJob                                    | null                                                           |          0 |   1548184502 |           0 |                  0 |
|     6 | OCA\Activity\BackgroundJob\EmailNotification                    | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
|     7 | OCA\Activity\BackgroundJob\ExpireActivities                     | null                                                           | 1548118802 |   1548170102 |           0 |                  0 |
|     8 | OCA\Federation\SyncJob                                          | null                                                           | 1548118802 |   1548170102 |           0 |                  0 |
|     9 | OCA\Files_Sharing\DeleteOrphanedSharesJob                       | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
|    10 | OCA\Files_Sharing\ExpireSharesJob                               | null                                                           | 1548118802 |   1548170102 |           0 |                  0 |
|    11 | OCA\UpdateNotification\Notification\BackgroundJob               | null                                                           | 1548118802 |   1548170102 |           0 |                 32 |
|    12 | OCA\Files_Versions\BackgroundJob\ExpireVersions                 | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
|    13 | \OC\Authentication\Token\DefaultTokenCleanupJob                 | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
|   888 | OCA\UpdateNotification\ResetTokenBackgroundJob                  | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
|  1009 | OCA\DAV\CardDAV\SyncJob                                         | null                                                           | 1516746613 |   1548184502 |           0 |                  2 |
|  1010 | OCA\NextcloudAnnouncements\Cron\Crawler                         | null                                                           | 1548118834 |   1548170102 |           0 |                 10 |
|  1111 | OC\BackgroundJob\Legacy\RegularJob                              | ["OCA\\News\\Cron\\Updater","run"]                             | 1548170102 |   1548170102 |           0 |                  0 |
|  1489 | OCA\Passman\BackgroundJob\ExpireCredentials                     | null                                                           | 1537861503 |   1548184502 |           0 |                  0 |
|  1576 | OCA\News\Cron\Updater                                           | null                                                           | 1537863301 |   1548194402 |           0 |                  1 |
|  2360 | OCA\Deck\Cron\DeleteCron                                        | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
|  2575 | OCA\QuotaWarning\Job\User                                       | {"uid":"user1"}                                                | 1548118834 |   1548170102 |           0 |                  0 |
|  2576 | OCA\QuotaWarning\Job\User                                       | {"uid":"user2"}                                                | 1548118834 |   1548170102 |           0 |                  0 |
|  2577 | OCA\QuotaWarning\Job\User                                       | {"uid":"user3"}                                                | 1548118834 |   1548170102 |           0 |                  0 |
|  2578 | OCA\QuotaWarning\Job\User                                       | {"uid":"user4"}                                                | 1548118834 |   1548170102 |           0 |                  0 |
|  2579 | OCA\QuotaWarning\Job\User                                       | {"uid":"user5"}                                                | 1548118834 |   1548170102 |           0 |                  0 |
|  2580 | OCA\QuotaWarning\Job\User                                       | {"uid":"user5"}                                                | 1548118834 |   1548170102 |           0 |                  0 |
|  2857 | OCA\QuotaWarning\Job\User                                       | {"uid":"user7"}                                                | 1548118834 |   1548170102 |           0 |                  0 |
|  7572 | OC\BackgroundJob\Legacy\RegularJob                              | ["OCA\\Music\\Backgroundjob\\CleanUp","run"]                   | 1548170102 |   1548170102 |           0 |                  0 |
|  7751 | OCA\OwnBackup\Jobs\BackupJob                                    | null                                                           | 1537863302 |   1548194402 |           0 |                  0 |
| 12494 | OCA\Deck\Cron\ScheduledNotifications                            | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
| 12555 | OC\Authentication\Token\DefaultTokenCleanupJob                  | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
| 12556 | OC\Log\Rotate                                                   | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
| 12559 | OCA\Spreed\BackgroundJob\ExpireSignalingMessage                 | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
| 12665 | OCA\DropIt\BackgroundJob\CleanUpJob                             | null                                                           | 1548118834 |   1548170102 |           0 |                  0 |
| 12712 | OCA\Spreed\BackgroundJob\RemoveEmptyRooms                       | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
| 12713 | OCA\Spreed\BackgroundJob\ResetInCallFlags                       | null                                                           | 1532545223 |   1548184502 |           0 |                  0 |
| 13132 | OCA\RecommendationAssistant\BackgroundJob\RecommenderJob        | null                                                           |          0 |   1548184502 |           0 |                  0 |
| 13571 | OC\Preview\BackgroundCleanupJob                                 | null                                                           | 1548166502 |   1548170102 |           0 |                  0 |
| 13573 | OCA\DAV\BackgroundJob\CleanupDirectLinksJob                     | null                                                           | 1548154802 |   1548170102 |           0 |                  0 |
| 13574 | OCA\DAV\BackgroundJob\UpdateCalendarResourcesRoomsBackgroundJob | null                                                           | 1548166502 |   1548170102 |           0 |                  0 |
| 13575 | OCA\DAV\BackgroundJob\CleanupInvitationTokenJob                 | null                                                           | 1548154802 |   1548170102 |           0 |                  0 |
| 13576 | OCA\Files_Sharing\BackgroundJob\FederatedSharesDiscoverJob      | null                                                           | 1548154802 |   1548170102 |           0 |                  0 |
| 13577 | OCA\Support\BackgroundJobs\CheckSubscription                    | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
| 16093 | OCA\TwoFactorNextcloudNotification\BackgroundJob\CleanupTokens  | null                                                           | 1547248502 |   1548158401 |  1548158401 |                  0 |
| 17277 | OCA\QuotaWarning\Job\User                                       | {"uid":"user7"}                                                | 1548121502 |   1548170102 |           0 |                  0 |
| 17736 | OCA\Deck\Cron\CardDescriptionActivity                           | null                                                           | 1548170102 |   1548170102 |           0 |                  0 |
| 17780 | OCA\GroupFolders\BackgroundJob\ExpireGroupVersions              | null                                                           |          0 |   1548189002 |           0 |                  0 |
| 17902 | OCA\DAV\BackgroundJob\RefreshWebcalJob                          | {"principaluri":"principals\/users\/user5","uri":"dummy"}      | 1547944226 |   1548170102 |           0 |                 19 |
+-------+-----------------------------------------------------------------+----------------------------------------------------------------+------------+--------------+-------------+--------------------+
48 rows in set (0.00 sec)
```

</details>
---

@jookk this one^ should probably sufficient; I don't think you need to search for yours too...

**Comment by jookk** [comment]
My oc_jobs table:
![screenshot_20190123-093428_1](https://user-images.githubusercontent.com/11217085/51593546-b0e18780-1ef2-11e9-90b4-0070f99052bc.jpg)

**Comment by MorrisJobke** [comment]
Fixed in #13607
